src/components/Button/SignInButton.jsx

```diff
@@ -0,0 +1,32 @@
+import { useNavigate } from "react-router-dom";
+
+import useBoundStore from "../../store/client/useBoundStore";
+import Button from "../UI/Button";
+
+const SignInButton = () => {
+  const navigate = useNavigate();
+
+  const isSignIn = useBoundStore((state) => state.isSignIn);
+  const asyncSignIn = useBoundStore((state) => state.asyncSignIn);
+  const signOut = useBoundStore((state) => state.signOut);
```

Comment:
원래 저 같은 경우는 원래 위와 같은 코드를 이렇게 적었습니다.
```js
const { isSignIn, asyncSignIn, signOut } = useBoundStore((state) => ({
  isSignIn: state.isSignIn,
  asyncSignIn: state.asyncSignIn,
  signOut: state.signOut,
}));
```

하지만 왜인지, 여러 개를 쓰지 않고 하나를 쓰더라도, 객체로 리턴하는 경우에 아래와 같은 에러가 뜨더라구요. 우선 에러를 없애기 위해 이것저것 해보다가 개별적으로 전역 상태를 가져오는 방법을 사용하니 에러가 해결되는 것을 확인하였는데요, 왜 에러가 생기는지의 이유는 찾지 못했습니다. 추후 한 번 연구해봐야할 사항인 것 같습니다.

![error](https://github.com/user-attachments/assets/69d6d844-9215-410e-9435-6d7b1a2beddb)

Reply:
**Zustand 공식 문서 참고**
- [Lifting state up, again](https://zustand.docs.pmnd.rs/guides/tutorial-tic-tac-toe#lifting-state-up,-again)
- [Slices-pattern](https://zustand.docs.pmnd.rs/guides/slices-pattern)

문서을 보면 useBoundState 전체에서 사용할 상태만 구조분해할당하거나 개별적으로 가져오는 형태들만 명시되어 있네요. 그래도 적어주신 방식을 막아두진 않았을 것 같은데 이상하네요..ㅎㅎ
저는 개별적으로 선언하는 방식이 더 명확한 것 같아서 그래도 유지해도 좋을 것 같습니다!

Reply:
그러게요,, 이전에는 계속 저렇게 썼는데 잘 되었었는데요, 버전 업이 되면서 막혔을지, 아니면 뭔가 R.Q.(React Query)랑 같이 쓰면서 뭔가 있는지... 이런저런 이유를 생각해보고 있는데, 설마... 싶네요.

답변 감사합니다!

Reply:
```suggestion
  const { isSignIn, asyncSignIn, signOut } = useBoundStore(
     useShallow((state) => ({
       isSignIn: state.isSignIn,
       asyncSignIn: state.asyncSignIn,
       signOut: state.signOut
     })),
```

Zustand에서 리렌더링 최적화 도와주는 `useShallow` 함수로 감싸주는 방법을 같이 검토해보면 좋을 것 같습니다. 우선 개발 서버에서 위 코드 수정해보니 에러가 해결되었는데, 처음 적용해본 것이다보니 같이 짚고 넘어가면 좋을 것 같아요.

현재 store에서 Boolean 외에 함수도 가져오는데, 요 부분으로 리렌더링 발생시키는 요인들이 있는 것 같아요.

참고
> In this case you might want to use useShallow to avoid a rerender if the computed value is always shallow equal the previous one. [prevent-rerenders-with-use-shallow.md](https://github.com/pmndrs/zustand/blob/main/docs/guides/prevent-rerenders-with-use-shallow.md)
- [Selecting multiple state slices](https://github.com/pmndrs/zustand?tab=readme-ov-file#selecting-multiple-state-slices)
- [최근 동일한 에러로 Zustand Maintainer가 `useShallow` 사용을 권장한 이슈](https://github.com/pmndrs/zustand/discussions/1936#discussioncomment-11025373)

Reply:
@jin-ttao 와...! 한 번 코드 상에 적용시켜 볼게요! state 내부에서 객체 내부 한 뎁스 이후에서 객체 전역 상태가 있어서 그랬을 수도 있겠다 싶네요.

Reply:
좋습니다! 현재 방식이 저희가 더 확신할 수 있는 방법이라 리스크가 적을 것 같아요. `useShallow`는 검증 후 반영해보아요.